app/assets/javascripts/components/buttons/generic-button-component.js

```diff
@@ -0,0 +1,41 @@
+ManageIQ.angular.app.component('genericButtonComponent', {
+  bindings: {
+    angularForm: '=',
+    modelCopy: '=',
+    model: '=',
+    redirectUrl: '=?',
+    newRecord: '<',
+    entity: '=?',
+    entityName: '=?',
+    onSaveClicked: '&?',
+    onAddClicked: '&?',
+  },
+  controllerAs: 'vm',
+  controller: genericButtonController,
+  templateUrl: '/static/buttons/generic_button_component.html.haml',
+});
+
+genericButtonController.$inject = ['miqService', '$window'];
+
+function genericButtonController(miqService) {
+  var vm = this;
+
+  vm.$onInit = function() {
+    vm.saveable = miqService.saveable;
+
+    vm.resetClicked = function() {
+      vm.model = angular.copy(vm.modelCopy );
+      vm.angularForm.$setUntouched(true);
+      vm.angularForm.$setPristine(true);
+      miqService.miqFlash('warn', __('All changes have been reset'));
+    };
+
+    vm.cancelClicked = function() {
+      if (vm.newRecord) {
```

Comment:
@himdel Hopefully this background on `generic-button-component` will help...

Do not think of it as *yet another* button component, because it's not.
It's a little more evolved than your basic button component definition.
It now caters to the needs of the form. In fact, it is really a child of the main form component, and hence sorta a form component too.

Maybe I should rename `generic-button-component` to `form-button-component` or something. 
I think, the right name will convey it's real purpose and what it actually does.
WDYT?